packages/server/test/integration/cypress_spec.js

```diff
@@ -338,8 +339,11 @@ describe('lib/cypress', () => {
       await clearCtx()
 
       sinon.stub(electron.app, 'on').withArgs('ready').yieldsAsync()
-      sinon.stub(runMode, 'waitForSocketConnection').resolves()
-      sinon.stub(runMode, 'listenForProjectEnd').resolves({ stats: { failures: 0 } })
+      globalThis.CY_TEST_MOCK = {
```

Comment:
So, I introduced this pattern because we have a lot of integration test coverage here that does deep mocking of (what was previously on) `module.exports`. I know this is an unfortunate way to handle mocking, but I felt it was the lesser of evils... the other options being to export everything to make it stubbable and call methods off `module.exports`, or some kinda proxyquire hackery. Open to better ideas here.

Reply:
I really question the value of some of these tests, where we mock 90% of the world to test some random method... I almost wonder if we collapse ~100 unit tests into ~5 system tests and just aggressively delete some of these. I think a lot of these are from the days prior to `system-tests` existing? 

If we have really complex unit level logic, stick it in a pure function and unit test that, so we don't need so much mocking/stubbing.

Reply:
Agreed. I didn't want to have to dive into figuring out what edge cases are/aren't covered in these integration tests versus other tests for this PR, but we do have way too many tests like this.